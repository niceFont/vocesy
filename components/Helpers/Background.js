const Background = props => {
	return (
		<div
			style={{
				position: "absolute",
				width: "100%",
				minHeight: "100%",
				backgroundColor: "white"
				/* 				background: "url('/static/background2.jpg')",
								backgroundSize: "cover",
								backgroundAttachment: "fixed",
								backgroundPosition: "center", */
			}}
		>
			{props.children}
		</div>
	)
}

export default Background
