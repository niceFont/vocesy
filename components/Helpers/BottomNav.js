import { Navbar, Nav } from "react-bootstrap"
import Link from "next/link"

const BottomNav = () => {
	return (
		<Navbar
			variant="dark"
			bg="dark"
			style={{
				position: "absolute",
				bottom: 0,
				width: "100%",
			}}>
			<Nav>
				<Nav.Item onClick={() => window.location.href = "mailto:peterwilms@hotmail.de"}>
					<Nav.Link as="a">
						Contact
					</Nav.Link>
				</Nav.Item>
				{/* 				<Nav.Item>
					<Link href="/about">
						<Nav.Link as="a">
							About
						</Nav.Link>
					</Link>
				</Nav.Item> */}
				<Nav.Item>
					<Link href="/privacy">
						<Nav.Link as="a">
							Privacy
						</Nav.Link>
					</Link>
				</Nav.Item>
				<Nav.Item>
					<Link href="/tos">
						<Nav.Link as="a">
							Terms of Service
						</Nav.Link>
					</Link>
				</Nav.Item>
				{/* <Nav.Item>
					<Link href="/cookies">
						<Nav.Link as="a">
							Cookies
						</Nav.Link>
					</Link>
				</Nav.Item> */}
			</Nav>
		</Navbar>
	)
}

export default BottomNav
