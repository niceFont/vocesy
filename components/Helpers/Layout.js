import {Nav,
	NavDropdown,
	Navbar,
	NavbarBrand} from "react-bootstrap"
import Link from "next/link"
import { ExtractName } from "../../lib/utils"

export const Layout = props => {
	return (
		<React.Fragment>
			<Navbar
				style={{
					paddingLeft: "4%",
					paddingRight: "4%",
					backgroundColor: "white",
					borderBottom: "1px solid lightgray"
				}}
				fixed="top"
				variant="light">
				<NavbarBrand>
					<Link href="/">
						<img
							style={{
								width: 120, height: 40 
							}}
							src="/static/logo.png"
						/>
					</Link>
				</NavbarBrand>
				<Nav>
					<Link href="/decks/create">
						<Nav.Link as="a">Create</Nav.Link>
					</Link>
					<Nav.Item>
						<Link href="/decks">
							<Nav.Link as="a">My Decks</Nav.Link>
						</Link>
					</Nav.Item>
				</Nav>
				<Navbar.Collapse className="justify-content-end">
					<Nav>
						{!props.user && (
							<Nav.Item>
								<Link href="/login" replace>
									<Nav.Link as="a">log in</Nav.Link>
								</Link>
							</Nav.Item>
						)}
						{props.user && (
							<div style={{
								display: "flex", flexDirection: "row"
							}}>
								<img style={{
									width: 30, height: 30, borderRadius: "100%", margin: "5px 0 5px 0"
								}} src={props.user.picture}></img>
								<NavDropdown
									title={ExtractName(props.user.displayName).split(" ").join("").toLowerCase()}
									id="nav-dropdown">
									<Link
										href={`/user/${encodeURI(ExtractName(props.user.displayName).split(" ").join("").toLowerCase())}`}>
										<NavDropdown.Item as="a">
										Profile
										</NavDropdown.Item>
									</Link>

									<Link href="/logout">
										<NavDropdown.Item
											style={{
												color: "#e84646" 
											}}
											as="a">
										Log Out
										</NavDropdown.Item>
									</Link>
								</NavDropdown>
							</div>
						)}
					</Nav>
				</Navbar.Collapse>
			</Navbar>
			{props.children}

			<Navbar fixed="bottom" bg="light" variant="dark">
				<Nav>

					<Nav.Item>
						<span>ver. 1.01a Still in Development...</span>
					</Nav.Item>
				</Nav>
			</Navbar>
		</React.Fragment>
	)
}
