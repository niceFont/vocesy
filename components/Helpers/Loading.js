import { Spinner, Container, Row } from "react-bootstrap"

export const Loading = props => {
	return (
		<div>
			{!props.fetched && (
				<Container>
					<Row className="justify-content-center">
						<Spinner size="lg" animation="border" role="status">
							<span className="sr-only">Loading...</span>
						</Spinner>
					</Row>
				</Container>
			)}
		</div>
	)
}
