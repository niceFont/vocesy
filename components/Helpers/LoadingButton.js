import {Button, Spinner} from "react-bootstrap"
import React from "react"


export default function LoadingButton({sending,...props}) {
	return (
		<Button {...props}>
			{sending ? <Spinner animation="border"></Spinner> : 
				<React.Fragment>
					{props.children}
				</React.Fragment>
			}
		</Button>
	)
}
