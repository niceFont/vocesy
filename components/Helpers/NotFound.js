
export const NotFound = () => {


	return (
		<h4> 404 | We could not find what you were looking for.</h4>
	)

}