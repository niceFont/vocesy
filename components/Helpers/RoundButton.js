import { Button } from "react-bootstrap"

const RoundButton = props => {
	return (
		<Button
			style={{
				...props.style,
				borderRadius: 25,
				padding: 0,
			}}
			{...props}
		>
			{props.children}
		</Button>
	)
}

export default RoundButton
