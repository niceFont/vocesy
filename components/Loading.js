import { Spinner } from "react-bootstrap"


export const Loading = (props) => {



	return (

		<div>

			{typeof window !== "undefined" ? <div>{props.children}</div>: <Spinner animation="border" role="status">
				<span className="sr-only">Loading...</span>
			</Spinner>}
		</div>

	)    
}