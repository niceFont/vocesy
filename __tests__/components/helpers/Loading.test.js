/*eslint no-undef: off*/
/*eslint camelcase: off*/
import renderer from "react-test-renderer"
import React from "react"
import { Loading } from "../../../components/Helpers/Loading"


describe("Loading", () => {
    
	it("should render loading", () => {
		const tree = renderer.create(<Loading fetched={false}></Loading>)
		expect(tree.toJSON()).toMatchSnapshot()
	})
    
})

