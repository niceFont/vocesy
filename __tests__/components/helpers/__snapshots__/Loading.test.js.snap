// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render loading 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="spinner-border spinner-border-lg"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
